app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/home/HomeViewController.kt

```diff
@@ -4,57 +4,152 @@ import android.os.Bundle
 import com.bluelinelabs.conductor.Controller
 import com.bluelinelabs.conductor.RouterTransaction
 import com.bluelinelabs.conductor.changehandler.HorizontalChangeHandler
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.SupervisorJob
-import kotlinx.coroutines.cancelChildren
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
 import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.get
-import kotlin.coroutines.CoroutineContext
 
-class HomeViewController : Controller(), KoinComponent, CoroutineScope {
+class HomeViewController : Controller(), KoinComponent {
 
   private val viewModel: HomeViewModel = get()
-  override val coroutineContext: CoroutineContext = Dispatchers.Main + SupervisorJob()
 
   override fun onCreateView(
     inflater: LayoutInflater,
     container: ViewGroup,
     savedViewState: Bundle?,
-  ): View = inflater.inflate(R.layout.home_view_controller, container, false)
+  ): View {
+    return ComposeView(container.context).apply {
+      setContent {
+        TrainingKotlinTheme() {
+          PokemonHomeContent(
+            viewModel = viewModel,
+            onSelectPokemon = {
+              router.pushController(
+                RouterTransaction.with(DetailViewController(pokemon = it))
+                  .popChangeHandler(HorizontalChangeHandler())
+                  .pushChangeHandler(HorizontalChangeHandler())
+              )
+            }
+          )
+        }
+      }
+    }
+  }
+}
 
-  override fun onAttach(view: View) {
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun PokemonHomeContent(viewModel: HomeViewModel, onSelectPokemon: (Pokemon) -> Unit) {
 
-    val presenter = HomeViewPresenter(view)
+  val state: HomeViewState by viewModel.states.collectAsState()
 
-    launch {
-      presenter.events.collect { event: HomeViewEvents ->
-        when (event) {
-          is HomeViewEvents.Refresh -> launch(Dispatchers.IO) { viewModel.getAllPokemon() }
-          is HomeViewEvents.ItemClick -> router.pushController(
-            RouterTransaction.with(DetailViewController(event.pokemon))
-              .pushChangeHandler(HorizontalChangeHandler())
-              .popChangeHandler(HorizontalChangeHandler())
+  Scaffold(
+    topBar = {
+      TopAppBar {
+        Text(
+          text = stringResource(id = R.string.pokemon),
+          style = MaterialTheme.typography.h6,
+        )
+      }
+    }
+  ) { padding ->
+    SwipeRefresh(
+      modifier = Modifier
+        .fillMaxSize()
+        .padding(padding),
+      state = rememberSwipeRefreshState(isRefreshing = state == Loading),
+      onRefresh = { with(viewModel) { launch { getAllPokemon() } } }
+    ) {
+      AnimatedContent(targetState = state) {
+        when (val _state: HomeViewState = state) {
```

Comment:
with `AnimatedContent` you no longer need to reassign this state to a local variable - the receiver is already a local variable the compiler can smart cast on 
```kotlin
AnimatedContent(targetState = state) { content ->
  when (content) {
    ..
  }
}
```